content/blog/recipes-0-1-14/index.Rmd

```diff
@@ -0,0 +1,132 @@
+---
+output: hugodown::hugo_document
+
+slug: recipes-0-1-14
+title: recipes 0.1.14
+date: 2020-10-20
+author: Max Kuhn
+description: >
+    A new version of the recipes package contains a signficant API update and 
+    some additional features 
+
+photo:
+  url: https://unsplash.com/photos/ToswmEekSFI
+  author: Holly Stratton
+
+# one of: "deep-dive", "learn", "package", "programming", or "other"
+categories: [package] 
+tags: [tidymodels, recipes]
+---
+
+```{r startup, include = FALSE}
+library(tidymodels)
+thm <- theme_bw() + 
+  theme(
+    panel.background = element_rect(fill = "transparent", colour = NA), 
+    plot.background = element_rect(fill = "transparent", colour = NA),
+    legend.background = element_rect(fill = "transparent", colour = NA),
+    legend.key = element_rect(fill = "transparent", colour = NA)
+  )
+theme_set(thm)
+```
+
+We're stoked to announce the release of [recipes](https://recipes.tidymodels.org) 0.1.14. recipes is an alternative method for creating and preprocessing design matrices that can be used for modeling or visualization. 
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("recipes")
+```
+
+You can see a full list of changes in the [release notes](https://recipes.tidymodels.org/news/index.html). There are some improvements and changes to talk about. 
+
+## An alternative to `juice()`
+
+Now that we have taught with recipes for a few years, we've realized that there is a lot of  confusion about the differences between `juice()` and `bake()`:
+
+* `juice(recipe)` returns the preprocessed training set (at very low computational costs).
+* `bake(recipe, new_data)` applies the recipe to any data (e.g. training, testing, unknowns, etc.)
+
+We were not able to find ways to make this distinction clear for many users. 
+
+How could we solve this issue? We decided to come up with a better alternative to `juice()`  that would be more intuitive. As a result,  _all_ applications of the recipe can now use `bake()`: 
+
+* `bake(recipe, new_data = some_data_set)` works as usual. 
+* `bake(recipe, new_data = NULL)` returns the preprocessed training set. 
+
+This is precedented in base R since many `predict()` methods re-predict the training set when the `newdata` argument is `NULL` or missing. Note that there is no default for `new_data`; you have to set it to `NULL` to get the training set. 
+
+We felt that this was the best API change that we could make. An external poll showed some agreement: 
+
+```{r poll, echo  = FALSE, fig.align='center', out.width="80%"}
+knitr::include_graphics("juice-poll.png")
+```
+
+`juice()`, which is still my favorite R function name of all time, will not be removed; you can still use it. However, we will not use it in training materials or documentation. 
```

Comment:
```suggestion
`juice()`, which is still my favorite R function name of all time, will not be removed; you can still use it. However, we will not use it in training materials or most documentation. 
```
Since it is still a documented function